### ISSUE TITLE
Not null bit field is translated as nullable boolean data type in C#

### ISSUE THREAD

**Issue by jaimestuardo** [ErikEJ/EFCorePowerTools#2122]
Hello...

If I have a field in a SQL Server 2019 table, for example, 

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/9195659/60469bfb-dc3c-4f23-9c83-ed5545c87370)

And apply reverse engineering, this is generated:

    public bool? EquipoHabilitado { get; set; }

Is this a bug or can I solve it by configuring something?

**Comment by ErikEJ** [comment]
It is by design - Duplicate of https://github.com/ErikEJ/EFCorePowerTools/issues/160#issuecomment-453286573 - there is an advanced option to disable this, see [the wiki](https://github.com/ErikEJ/EFCorePowerTools/wiki/Reverse-Engineering#code-generation)

**Comment by jaimestuardo** [comment]
The solution I used was to remove default value for the bit feld. It is not clear for me why when setting a default value of "1", a nullable type is generated. 

I thought default values were only used when inserting records when that field is not specified. In EF, all fields are specified always when inserting or updating, so there is no possibility to pass a null value there.

**Comment by ErikEJ** [comment]
Did you read the linked issues?

**Comment by jaimestuardo** [comment]
Yes.... but I don't understand this sentence: "If the property wasn't nullable, there would be no way to store a 0.". The question is, why? If the property is not nullable, default bool value should be stored (false or zero).

As I said, EF always specify all fields explicitly when inserting records (when default value makes sense). Since EF always specifies all fields, always a 0 or a 1 will be passed. Never a null.

So, when reverse engineering the database using power tools, default value should not be taken into account. As I also said, when I removed default value from SQL Server table definition, EF Power Tools generated the field as `bool`, as expected.

**Comment by ErikEJ** [comment]
@jaimestuardo Agree, it is complex and confusing - have you tried with EF Core 8?
